### PR TITLE
parser: skip restoring schema name of CTE table columns

### LINKED ISSUE #33965: "references invalid table" when querying CTE views
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

```sql
-- set up 2 same tables in different db.
create database if not exists test;
create database if not exists test1;
drop table if exists test.t, test1.t;
drop view if exists test.v;
create table test.t (a int);
insert into test.t values (1);
create table test1.t (a int);
insert into test1.t values (2);

use test;
create view test.v as with tt as (select * from t) select * from tt;
use test;
select * from test.v;
use test1;
select * from test.v;
```

### 2. What did you expect to see? (Required)

```sql
mysql> use test;
Database changed
mysql> select * from test.v;
+------+
| a    |
+------+
|    1 |
+------+
1 row in set (0.01 sec)

mysql> use test1;
Database changed
mysql> select * from test.v;
+------+
| a    |
+------+
|    1 |
+------+
1 row in set (0.00 sec)
```

### 3. What did you see instead (Required)

```sql
mysql> use test;
Database changed
mysql> select * from test.v;
+------+
| a    |
+------+
|    1 |
+------+
1 row in set (0.01 sec)

mysql> use test1;
Database changed
mysql> select * from test.v;
ERROR 1356 (HY000): View 'test.v' references invalid table(s) or column(s) or function(s) or definer/invoker of view lack rights to use them
```

### 4. What is your TiDB version? (Required)

45e48b64989

### PR DESCRIPTION
**Pull request by tangenta**
### What problem does this PR solve?

Issue Number: close #33965

Problem Summary:

See https://github.com/pingcap/tidb/issues/33965#issuecomment-1099289856

```sql
create view test.v as with tt as (select * from t) select * from tt;
```

This SQL should be restored to 

```sql
WITH `tt` AS (SELECT `test`.`t`.`a` AS `a` FROM `test`.`t`) SELECT `tt`.`a` AS `a` FROM `tt`
                                                                   -- no schema name here
```
instead of
```sql
WITH `tt` AS (SELECT `test`.`t`.`a` AS `a` FROM `test`.`t`) SELECT `test`.`tt`.`a` AS `a` FROM `tt`
```

### What is changed and how it works?

This PR prevent restoring the schema name of a column, if this column comes from a CTE table.

### Check List

Tests 

- [ ] Unit test
- [x] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: executor/cte_test.go [hunk 1]
```diff
@@ -440,3 +441,28 @@ func TestCTEExecError(t *testing.T) {
...
 	}
 }
+
+// https://github.com/pingcap/tidb/issues/33965.
+func TestCTEsInView(t *testing.T) {
+	store, clean := testkit.CreateMockStore(t)
+	defer clean()
+	tk := testkit.NewTestKit(t, store)
+	tk.MustExec("use test;")
+	require.True(t, tk.Session().Auth(&auth.UserIdentity{Username: "root", Hostname: "%"}, nil, nil))
```
Comment (wjhuang2016) [inline executor/cte_test.go:451]: Why do you want to check this?

Comment (tangenta) [reply executor/cte_test.go:451]: It gives the current session a privilege so that `SELECT` views can pass the check.

**Review by xiongjiwei — APPROVED** [review]

**Review by wjhuang2016 — APPROVED** [review]

**Pull request merged**